CustomCraftSML/Serialization/Entries/CustomFood.cs

```diff
@@ -0,0 +1,298 @@
+            set => decayrate.Value = value;
+        }
+
+        public bool AllowOverfill
+        {
+            get => allowOverfill.Value;
+            set => allowOverfill.Value = value;
+        }
+
+        internal bool Decomposes => this.DecayRateMod > 0f;
+
+        protected static List<EmProperty> CustomFoodProperties => new List<EmProperty>(AliasRecipeProperties)
+        {
+            new EmProperty<FoodModel>(FoodModelKey, FoodModel.None) { Optional = true },
+            new EmProperty<TechType>(SpriteItemIdKey, TechType.None) { Optional = true },
+            new EmProperty<short>(FoodKey, 0) { Optional = false },
+            new EmProperty<short>(WaterKey, 0) { Optional = false },
+            new EmProperty<float>(DecayRateKey, 0) { Optional = true },
+            new EmYesNo(OverfillKey, true) { Optional = true },
+        };
+
+        internal TechType FoodPrefab
+        {
+            get
+            {
+                if (this.FoodType == FoodModel.None)
+                {
+                    if (this.FoodValue >= this.WaterValue)
+                    {
+                        return TechType.NutrientBlock;
+                    }
+                    else
+                    {
+                        return TechType.FilteredWater;
+                    }
+                }
+
+                return (TechType)this.FoodType;
+            }
+        }
+
+        internal string IconName
+        {
+            get
+            {
+                string imagePath = IOPath.Combine(FileLocations.AssetsFolder, $"{this.ItemID}.png");
+
+                if (File.Exists(imagePath))
+                    return imagePath;
+
+                if (this.FoodValue >= this.WaterValue)
+                    return "cake.png";
+                else
+                    return "juice.png";
+            }
+        }
+
+        public CustomFood() : this(TypeName, CustomFoodProperties)
+        {
+        }
+
+        protected CustomFood(string key) : this(key, CustomFoodProperties)
+        {
+        }
+
+        protected CustomFood(string key, ICollection<EmProperty> definitions) : base(key, definitions)
+        {
+            foodValue = (EmProperty<short>)Properties[FoodKey];
+            waterValue = (EmProperty<short>)Properties[WaterKey];
+            decayrate = (EmProperty<float>)Properties[DecayRateKey];
+            allowOverfill = (EmYesNo)Properties[OverfillKey];
+
+            techGroup.Value = TechGroup.Survival;
+            techCategory.DefaultValue = TechCategory.CookedFood;
+            foodModel = (EmProperty<FoodModel>) Properties[FoodModelKey];
+        }
+
+        internal override EmProperty Copy()
+        {
+            return new CustomFood(this.Key, this.CopyDefinitions);
+        }
+
+        public override bool PassesPreValidation()
+        {
+            return base.PassesPreValidation() & ValidateCustomFoodValues();
+        }
+
+        private bool ValidateCustomFoodValues()
```

Comment:
Additional validation to add to `ValidateCustomFoodValues`:  
A valid `CustomFood` entry should have a non-zero value for either `FoodValue` or `WaterValue`.  
If both of them are zero, then the food item has no function and should be considered invalid.  
```CSharp
if (this.FoodValue == 0 && this.WaterValue == 0)
   // Invalid entry
```